context/context.go

```diff
@@ -0,0 +1,141 @@
+package context
+
+import (
+	gocontext "context"
+	"fmt"
+
+	"github.com/flanksource/commons/logger"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type ContextOptions func(*Context)
+
+func WithTraceFn(fn func(*Context) bool) ContextOptions {
+	return func(opts *Context) {
+		opts.isTraceFn = fn
+	}
+}
+
+func WithDebugFn(fn func(*Context) bool) ContextOptions {
+	return func(opts *Context) {
+		opts.isDebugFn = fn
+	}
+}
+
+func WithTracer(tracer trace.Tracer) ContextOptions {
+	return func(opts *Context) {
+		opts.tracer = tracer
+	}
+}
+
+func WithLogger(log logger.Logger) ContextOptions {
+	return func(opts *Context) {
+		opts.logger = log
+	}
+}
+
+func NewContext(opts ...ContextOptions) *Context {
+	ctx := Context{
+		Context: gocontext.Background(),
+	}
+	for _, opt := range opts {
+		opt(&ctx)
+	}
+	if ctx.logger == nil {
+		ctx.logger = logger.StandardLogger()
+	}
+	if ctx.isDebugFn == nil {
+		ctx.isDebugFn = func(*Context) bool {
+			return ctx.logger.IsDebugEnabled()
+		}
+	}
+	if ctx.isTraceFn == nil {
+		ctx.isTraceFn = func(*Context) bool {
+			return ctx.logger.IsTraceEnabled()
+		}
+	}
+	if ctx.tracer == nil {
+		ctx.tracer = trace.NewNoopTracerProvider().Tracer("noop")
+	}
+	return &ctx
+}
+
+type Context struct {
+	gocontext.Context
+	logger    logger.Logger
+	isDebugFn func(*Context) bool
+	isTraceFn func(*Context) bool
+	tracer    trace.Tracer
```

Comment:
I think tracers are supposed to be package level vars and not a request scoped var. So, I don't think it should be the context.

_Example_:
Layers: Controller => API => DB
Each of these packages should have their own package-level tracer so the layers are visible